src/engine/currencyEngine.js

```diff
@@ -624,7 +624,7 @@ export class CurrencyEngine {
       }
       return edgeTransaction
     } catch (e) {
-      if (e.type === 'FundingError') throw new Error('InsufficientFundsError')
+      if (e.type === 'FundingError') throw new InsufficientFundsError()
```

Comment:
The `InsufficientFundsError` accepts a currencyCode as its one parameter. If you can add that in, we can merge this.